mpi.h

```diff
@@ -296,25 +296,25 @@ enum {
 #define MPI_UNWEIGHTED      ((int*)4)
 #define MPI_WEIGHTS_EMPTY   ((int*)5)
 
-// Other constants
-#define MPI_BSEND_OVERHEAD                   512 // MPICH=96, OMPI=128
-
-// String size constants
-#define MPI_MAX_DATAREP_STRING               128 // MPICH=OMPI=128 (MPICH has it in `mpio.h`)
-#define MPI_MAX_ERROR_STRING                 512 // MPICH was bigger
-#define MPI_MAX_INFO_KEY                     256 // MPICH was bigger
-#define MPI_MAX_INFO_VAL                    1024 // MPICH was bigger
-#define MPI_MAX_LIBRARY_VERSION_STRING      8192 // MPICH was bigger
-#define MPI_MAX_OBJECT_NAME                  128 // MPICH was bigger
-#define MPI_MAX_PORT_NAME                   1024 // OMPI was bigger
-#define MPI_MAX_PROCESSOR_NAME               256 // OMPI was bigger
-#define MPI_MAX_STRINGTAG_LEN               1024 // OMPI was bigger (v5.0+)
-#define MPI_MAX_PSET_NAME_LEN                512 // OMPI was bigger (v5.0+)
-
-// Mode Constants
-// must be powers-of-2 to support OR-ing
+// Other Constants
 enum {
-    // Files
+    MPI_BSEND_OVERHEAD                  = 512 // MPICH=96, OMPI=128
+};
+
+// Maximum Sizes for Strings
+enum {
+    MPI_MAX_DATAREP_STRING              =  128, // MPICH=OMPI=128 (MPICH has it in `mpio.h`)
+    MPI_MAX_ERROR_STRING                =  512, // MPICH was bigger
+    MPI_MAX_INFO_KEY                    =  256, // MPICH was bigger
+    MPI_MAX_INFO_VAL                    = 1024, // MPICH was bigger
+    MPI_MAX_LIBRARY_VERSION_STRING      = 8192, // MPICH was bigger
+    MPI_MAX_OBJECT_NAME                 =  128, // MPICH was bigger
+    MPI_MAX_PORT_NAME                   = 1024, // OMPI was bigger
+    MPI_MAX_PROCESSOR_NAME              =  256, // OMPI was bigger
+    MPI_MAX_STRINGTAG_LEN               = 1024, // OMPI was bigger (v5.0+)
+    MPI_MAX_PSET_NAME_LEN               =  512  // OMPI was bigger (v5.0+)
+};
```

Comment:
Enum values may not work in C preprocessor -- https://stackoverflow.com/questions/34677148/why-does-the-c-preprocessor-consider-enum-values-as-equal

Reply:
Definitely true, enums do not work at all at the C preprocessor level. Why should the `mpi.h` header define this particular set of size-related constants in preprocessor rather than enum as all the others? Note that the CI tests for this PR built mpi4py just fine. What am I missing?

Reply:
I did this change for consistency. Everything is an enum now, except MPI_VERSION/SUBVERSION, in this case preprocessor values may be used for conditional compilation.

Reply:
I think it is common to use the preprocessor to check limits to decide whether a certain array fits in a static storage or to use heap allocation.

Reply:
I understand and agree with the rationale when you consider an API and multiple MPI implementations. However, within the ABI constraints, these size values are known in advance and guaranteed to be the same across implementations. Therefore, I thought that breaking the rules for these constants was not worth it.

Anyway, as we envision some time where user codes will have to deal with with the old stuff and the new stuff, I'll revert the change and use preprocessor constants for the string sizes.

I'm not sure whether the same rationale apply to `MPI_BSEND_OVERHEAD` and its intended use. I would not expect attached buffers to be small statically-allocated buffers in the stack. However, to be in the safe side, and as the constant is also a size, I'll make it a `#define` too for now.  But maybe we should not?